components/login/login_controller/login_controller.jsx

```diff
@@ -553,17 +553,18 @@ class LoginController extends React.Component {
             );
 
             if (this.state.loading) {
-                loginButton =
-                (<span>
-                    <span
-                        className='fa fa-refresh icon--rotate'
-                        title={Utils.localizeMessage('generic_icons.loading', 'Loading Icon')}
-                    />
-                    <FormattedMessage
-                        id='login.signInLoading'
-                        defaultMessage='Signing in...'
-                    />
-                </span>);
+                loginButton = (
+                    <span data-cy='login_signing_in'>
```

Comment:
do we need to add `data-cy` to everything that we want to test? can you explain this a bit?

Reply:
Ref: https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements

It is the recommended best practice in selecting an element.  I'm particularly concern about "selectors break from development changes" which I think we'll definitely hit us from time to time.  

For most cases, I think we're good at using element ID but for us to be more careful on the change we'll be making, seeing `data-*` will mean that such element is important and is being *watched*.
Most likely, we'll be adding `data-*` to the following:
- a page, to easily identify if it has been loaded
- a section, like RHS, modal, etc
- a dynamic component/element
- an element being used in transition
- an element that addresses or has history of bug